pandas_gbq/tests/test_gbq.py

```diff
@@ -1422,6 +1422,48 @@ def test_schema_is_subset_fails_if_not_subset(self):
         assert self.sut.schema_is_subset(
             dataset, table_name, tested_schema) is False
 
+    def test_upload_data_with_valid_user_schema(self):
+        # Issue #46; tests test scenarios with user-provided
+        # schemas
+        df = tm.makeMixedDataFrame()
+        test_id = "15"
+        test_schema = [{'name': 'A', 'type': 'FLOAT'},
```

Comment:
There's a chance this might fail with version 0.29.0 of `google-cloud-bigquery` due to https://github.com/GoogleCloudPlatform/google-cloud-python/issues/4456

Reply:
Generated schemas do not include the `mode` property in the fields either. So this should be fine.